### PR TITLE
Get rid of high connection latency

### PR DESCRIPTION
**Pull request by asvetlov**
Fix for #448

### CODE REVIEW

File: aiohttp/server.py [hunk 1]
```diff
@@ -291,15 +291,20 @@ def start(self):
...
-                            'Start keep-alive timer for %s sec.',
-                            self._keep_alive_period)
-                        self._keep_alive_handle = self._loop.call_later(
-                            self._keep_alive_period, self.transport.close)
-                    elif self._keep_alive and self._keep_alive_on:
-                        # do nothing, rely on kernel or upstream server
-                        pass
+                    if self._keep_alive:
+                        sock = self.transport.get_extra_info('socket')
+                        sock.setsockopt(socket.IPPROTO_TCP,
```
Comment (jashandeep-sohi) [inline aiohttp/server.py]: Might want to make sure a socket was actually returned and the type of the socket is TCP.

Comment (asvetlov) [inline aiohttp/server.py]: good point

**Comment by asvetlov** [comment]
Added `set_tcp_nodelay` method, more tests

**Comment by asvetlov** [comment]
@fafhrd91 are you agree with PR?

**Comment by fafhrd91** [comment]
i still think tcp_nodelay should be on by default. otherwise i am fine

File: aiohttp/server.py [hunk 2]
```diff
@@ -388,3 +391,19 @@ def handle_request(self, message, payload):
...
+
+    def set_tcp_nodelay(self, nodelay):
+        if nodelay == self._tcp_nodelay:
+            return
+        self._tcp_nodelay = nodelay
+        if self.transport is None:
+            # transport is closed
+            return
+        sock = self.transport.get_extra_info('socket')
+        if sock.family not in (socket.AF_INET, socket.AF_INET6):
```
Comment (jashandeep-sohi) [inline aiohttp/server.py:407]: Transports need not have a `socket`, i.e. `get_extra_info('socket')` might return a `None`.
So, should probably do a `if sock and sock.family not in (socket.AF_INET, socket.AF_INET6)` to be on the safe side.

Comment (socketpair) [inline aiohttp/server.py:407]: @asvetlov I agree with @jashandeep-sohi

File: aiohttp/server.py [hunk 3]
```diff
@@ -388,3 +391,19 @@ def handle_request(self, message, payload):
...
+            return
+        sock.setsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY, 1)
```
Comment (socketpair) [inline aiohttp/server.py:409]: Replace `1` with `1 if nodelay else 0`

File: aiohttp/server.py [hunk 4]
```diff
@@ -388,3 +391,19 @@ def handle_request(self, message, payload):
...
+        return self._tcp_nodelay
+
+    def set_tcp_nodelay(self, nodelay):
+        if nodelay == self._tcp_nodelay:
+            return
+        self._tcp_nodelay = nodelay
+        if self.transport is None:
+            # transport is closed
+            return
+        sock = self.transport.get_extra_info('socket')
```
Comment (socketpair) [inline aiohttp/server.py:406]: Also it will be nice to cache socket reference in order to speedup setting and removing nodelay

**Comment by asvetlov** [comment]
Close in favor of #680 
All comments are satisfied in the superseeder.

**Pull request closed without merging**